amf-shapes/shared/src/main/scala/amf/shapes/client/scala/model/domain/jsonldinstance/JsonLDObject.scala

```diff
@@ -22,4 +24,68 @@ class JsonLDObject(
 
   /** Value , path + field value that is used to compose the id when the object its adopted */
   override def componentId = s"/${path.toString}"
+
+  implicit class FieldBuilder(property: String) {
+
+    def toObjField(meta: Obj): Field = Field(meta, ValueType(property))
+
+    def toObjListField: Field = Field(Type.Array(Type.ObjType), ValueType(property))
+    def toStrField: Field     = Field(Type.Str, ValueType(property))
+
+    def toStrListField: Field = Field(Type.Array(Type.Str), ValueType(property))
+
+    def toIntField: Field = Field(Type.Int, ValueType(property))
+
+    def toIntListField: Field = Field(Type.Array(Type.Int), ValueType(property))
+    def toBoolField: Field    = Field(Type.Bool, ValueType(property))
+
+    def toBoolListField: Field = Field(Type.Array(Type.Bool), ValueType(property))
+    def toFloatField: Field    = Field(Type.Float, ValueType(property))
+
+    def toFloatListField: Field = Field(Type.Array(Type.Float), ValueType(property))
+  }
+
+  private def buildString(value: String) = new JsonLDScalar(value, DataTypes.String)
+
+  private def buildInteger(value: Int) = new JsonLDScalar(value, DataTypes.Integer)
+
+  private def buildBoolean(value: Boolean) = new JsonLDScalar(value, DataTypes.Boolean)
+
+  private def buildFloat(value: Float) = new JsonLDScalar(value, DataTypes.Float)
+
+  private def buildArray(values: Seq[JsonLDElement]) = JsonLDArray(values)
+
+  private def updateModel(field: Field) = model.copy(fields = model.fields.filterNot(_ == field) :+ field)
+  private def copyWithProperty(field: Field, element: JsonLDElement) =
+    copy(fields = fields.copy(), model = updateModel(field)).set(field, element)
+
+  def withProperty(property: String, value: String): JsonLDObject =
+    copyWithProperty(property.toStrField, buildString(value))
+
+  def withProperty(property: String, value: Int): JsonLDObject =
+    copyWithProperty(property.toIntField, buildInteger(value))
+
```

Comment:
same as last comment

Reply:
Didn't understood, sorry. In this case the method name is the same for all scalar, the type of the "value" param is the only difference, but that's to avoid do an internal match from "any" (and be forced to handle the unrecognized types)